src/kaczmarz/_variants.py

```diff
@@ -3,10 +3,25 @@
 from collections import deque
 
 import numpy as np
+from scipy import sparse
 
 import kaczmarz
 
 
+def scale_rows(A, v):
+    return sparse.spdiags(v, 0, len(v), len(v)) @ A
+
+
+def scale_cols(A, v):
+    return A @ sparse.spdiags(v, 0, len(v), len(v))
+
+
+def square(A):
+    if sparse.issparse(A):
+        return A.power(2)
+    return A ** 2
```

Comment:
These functions need docstrings! And please move them to a new file `_utils.py` and import them at the top with
```python
from ._utils import scale_rows, scale_cols, square
```

Reply:
If you can think of a better (more specific/informative) name for the new file, please use it!